apps/communications/dialer/test/unit/keypad_test.js

```diff
@@ -678,6 +681,87 @@ suite('dialer/keypad', function() {
         sinon.assert.notCalled(CallHandler.call);
       });
     });
+
+    suite('Speed dial', function() {
+      var speedDialNum = '1#';
+
+      setup(function() {
+        subject._phoneNumber = '';
+      });
+
+      test(speedDialNum + ' is a speed dial number', function() {
+        assert.isTrue(subject._isSpeedDialNumber(speedDialNum));
+      });
+
+      test('123 is not a speed dial number', function() {
+        assert.isFalse(subject._isSpeedDialNumber('123'));
+      });
+
+      test('*#31# is not a speed dial number', function() {
+        assert.isFalse(subject._isSpeedDialNumber('123'));
+      });
+
+      test('Starts speed dial upon typing ' + speedDialNum, function() {
+        var node;
+        var fakeEventStart;
+        var fakeEventEnd;
+
+        node = document.createElement('div');
+        fakeEventStart = {
+          target: node,
+          stopPropagation: function() {},
+          type: 'touchstart'
+        };
+        fakeEventEnd = {
+          target: node,
+          stopPropagation: function() {},
+          type: 'touchend'
+        };
+
+        this.sinon.stub(KeypadManager, '_getSpeedDialNumber', function() {
+          return Promise.resolve('123');
+        });
+
+        for (var i = 0, end = speedDialNum.length; i < end; i++) {
+          fakeEventStart.target.dataset.value = speedDialNum.charAt(i);
+          subject.keyHandler(fakeEventStart);
+          fakeEventEnd.target.dataset.value = speedDialNum.charAt(i);
+          subject.keyHandler(fakeEventEnd);
+        }
+
+        sinon.assert.calledWith(KeypadManager._getSpeedDialNumber, 1);
+      });
+
+      suite('Getting a speed dial number', function() {
+        var speedDialIndex = '1';
+        var number = '123';
+
+        suiteSetup(function() {
+          navigator.mozIccManager.adnContacts = [
+            {
+              id: '123456',
+              tel: [ { value: number } ]
+            }
+          ];
+        });
+
+        test('The overlay is displayed and then hidden', function(done) {
+          this.sinon.spy(ConfirmDialog, 'show');
+          this.sinon.spy(ConfirmDialog, 'hide');
+
+          subject._getSpeedDialNumber(speedDialIndex).then(function() {
+            sinon.assert.calledOnce(ConfirmDialog.show);
+            sinon.assert.calledOnce(ConfirmDialog.hide);
+          }).then(done, done);
+        });
+
+        test('The proper number is returned', function(done) {
+          subject._getSpeedDialNumber(speedDialIndex).then(function(number) {
```

Comment:
nit: maybe rename this variable number to something else to avoid confusion with the variable number above it.